pkg/acme/provider.go

```diff
@@ -0,0 +1,57 @@
+package acme
+
+import (
+	"github.com/go-acme/lego/v4/challenge"
+	"github.com/go-acme/lego/v4/challenge/dns01"
+	"github.com/loopholelabs/certifier/pkg/options"
+	"github.com/loopholelabs/certifier/pkg/storage"
+	"github.com/loopholelabs/logging"
+)
+
+var _ challenge.Provider = (*provider)(nil)
```

Comment:
Could you explain what is going on here?

Reply:
This line creates an empty variable that tells the compiler that *provider should fulfill the challenge.Provider interface.

It's basically a compiler check that we are fulfilling the interface